### DISCUSSION TITLE
Preferred email address with GitHub

### DISCUSSION THREAD

**Discussion by hendrikmaus** [dexidp/dex#2172] [Q&A] [answered]
How does Dex decide which email address to use when authenticating with GitHub? Is it always the primary address or can this be somehow picked by a user?

I am wondering how to let people choose which email address to expose if they login into [argo-cd](https://argoproj.github.io/argo-cd/operator-manual/user-management/#2-configure-argo-cd-for-sso) which we integrated with Dex and GitHub.

Maybe this is not even related to Dex, but to the application using it, so argo-cd in this case?

I'd appreciate some advice.

**Comment by nabokihms** [comment, marked as answer]
As for now, Dex has no feature to customize which email is to be picked. Dex picks the verified primary email (event if it is private).
You can not change this behavior because this logic is hardcoded.
